Fixes 119: fix some tests failing on seeded database

avisiedo opened the pull request:
- Update TestUpdateEmpty to make it more predictable.
- Fix TestSavePublicUrls test.
- Update TestUpdateBlank to make it more predictable.
- Update TestBulkCreateOneFails to make it more predictable.

Signed-off-by: Alejandro Visiedo <avisiedo@redhat.com>

avisiedo commented on pkg/dao/repository_test.go:
```diff
@@ -256,24 +272,44 @@ func (suite *RepositorySuite) TestUpdateEmpty() {
-	assert.Nil(t, err)
-	found := models.RepositoryConfiguration{}
-	suite.tx.Where("org_id = ?", org_id).First(&found)
+	// Create a RepositoryConfiguration record
+	repoPublic := repoPublicTest.DeepCopy()
+	err = tx.Create(&repoPublic).Error
+	require.NoError(t, err)
 
+	repoConfig := repoConfigTest1.DeepCopy()
+	repoConfig.RepositoryUUID = repoPublic.UUID
+	repoConfig.OrgID = org_id
+	err = tx.Create(&repoConfig).Error
```
I add the records without SeedRepositoryConfigurations to have control about the uuid to check later the records.

avisiedo commented on pkg/dao/repository_test.go:
```diff
@@ -256,24 +272,44 @@ func (suite *RepositorySuite) TestUpdateEmpty() {
+
+	// Retrieve the just created RepositoryConfiguration record
+	found := models.RepositoryConfiguration{}
+	err = tx.Where("org_id = ?", org_id).First(&found, "uuid = ?", repoConfig.UUID).Error
```
I add the primary key to First method to recover the specific record.

avisiedo commented on pkg/dao/repository_test.go:
```diff
@@ -256,24 +272,44 @@ func (suite *RepositorySuite) TestUpdateEmpty() {
 
-	suite.tx.First(&found)
+	// Check the updated data
+	tx.First(&found, "uuid = ?", repoConfig.UUID)
```
I add the primary key to First method to recover the specific record.

avisiedo commented on pkg/dao/repository_test.go:
```diff
@@ -352,12 +388,19 @@ func (suite *RepositorySuite) TestUpdateBlank() {
-	orgID := "900023"
+	orgID := orgIdTest
+
+	repo := repoPublicTest.DeepCopy()
+	err = tx.Create(&repo).Error
+	require.NoError(t, err)
+
+	repoConfig := repoConfigTest1.DeepCopy()
+	repoConfig.RepositoryUUID = repo.UUID
+	err = tx.Create(&repoConfig).Error
+	require.NoError(t, err)
```
add records by hand to have better control of the records to be recovered later.

avisiedo commented on pkg/dao/repository_test.go:
```diff
@@ -352,12 +388,19 @@ func (suite *RepositorySuite) TestUpdateBlank() {
-	assert.Nil(t, err)
 	found := models.RepositoryConfiguration{}
-	suite.tx.Preload("Repository").First(&found)
+	err = tx.Preload("Repository").First(&found, "uuid = ?", repoConfig.UUID).Error
```
Added specific primary key to be recovered.

avisiedo commented on pkg/dao/suite_test.go:
```diff
@@ -45,7 +45,7 @@ type PublicRepositorySuite struct {
 const accountIdTest = "817342"
 
-var repoTest1 = models.Repository{
+var repoPublicTest = models.Repository{
```
Only a renaming for clarity about it

jlsherrill commented:
https://issues.redhat.com/browse/HMSCONTENT-119

rverdile commented on pkg/dao/repository_test.go:
```diff
@@ -195,19 +195,19 @@ func (suite *RepositorySuite) TestBulkCreateOneFails() {
 
-	orgID := "1"
-	accountID := "2"
+	orgID := orgIdTest
```
should this orgID, and any remaining hard-coded orgIDs, be `seeds.RandomOrgId()`?  I know it's very unlikely, but what if the existing database has a repository the orgID `1` and URL `repo_1_url` already?

My understanding is, since the tests no longer wipe the database first, we depend on the OrgID to keep the test data separate.

rverdile commented on pkg/dao/repository_test.go:
```diff
@@ -216,15 +216,31 @@ func (suite *RepositorySuite) TestBulkCreateOneFails() {
+	var count int64
+	foundRepoConfig := []models.RepositoryConfiguration{}
+	err = tx.Model(&models.RepositoryConfiguration{}).
+		Where("repositories.url in (?)", urls).
```
We may want to also filter by OrgID here and anywhere else that isn't querying by UUID

rverdile reviewed (commented):
looking good, but there's some merge conflicts now

avisiedo commented:
I have realized more places to add the org_id filter, now it should be better. I have added at some points `require` instead of `assert` specially when the data (error or data models) are used later.

I have rebased the pr so now it should be in better shape

rverdile reviewed (approved):
lgtm

avisiedo commented:
/retest

jlsherrill commented:
@avisiedo feel free to go ahead and merge, the failing tests will be fixed by https://github.com/content-services/content-sources-backend/pull/86
